# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../vmailmgr/stricter_gcc.patch
# Copyright (C) 2004 - 2006 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---
--- ./lib/mystring/mystring.h.orig	2004-05-07 09:43:14.000000000 -0400
+++ ./lib/mystring/mystring.h	2004-05-07 09:43:29.000000000 -0400
@@ -18,6 +18,7 @@
 #define MYSTRING__H__
 
 #include <sys/types.h>
+#include <string.h>
 #include "mystring/rep.h"
 
 class mystringjoin;
--- ./cgi/Makefile.am.orig	2004-05-07 10:31:42.000000000 -0400
+++ ./cgi/Makefile.am	2004-05-07 10:32:34.000000000 -0400
@@ -3,7 +3,7 @@
 cgi_PROGRAMS = vpasswd vadduser vdeluser vaddalias vchattr vchforwards \
 	listvdomain
 
-CXXLINK = $(CC) $(CXXFLAGS) $(LDFLAGS) -o $@
+CXXLINK = $(CXX) $(CXXFLAGS) $(LDFLAGS) -o $@
 INCLUDES = -I../lib
 
 vpasswd_SOURCES = vpasswd.cc
--- ./commands/Makefile.am.orig	2004-05-07 10:31:53.000000000 -0400
+++ ./commands/Makefile.am	2004-05-07 10:32:41.000000000 -0400
@@ -31,7 +31,7 @@
 %.1: %.pod ../configure.in; $(POD2MAN) --section=1 $< >$@
 %.html: %.pod; $(POD2HTML) $< >$@
 
-CXXLINK = $(CC) $(CXXFLAGS) $(LDFLAGS) -o $@
+CXXLINK = $(CXX) $(CXXFLAGS) $(LDFLAGS) -o $@
 INCLUDES = -I../lib -DBINDIR=\"@bindir@\"
 
 listvdomain_SOURCES = listvdomain.cc vcommand.cc
--- ./daemon/Makefile.am.orig	2004-05-07 10:32:03.000000000 -0400
+++ ./daemon/Makefile.am	2004-05-07 10:32:48.000000000 -0400
@@ -20,7 +20,7 @@
 %.8: %.pod ../configure.in; $(POD2MAN) --section=8 $< >$@
 %.html: %.pod; $(POD2HTML) $< >$@
 
-CXXLINK = $(CC) $(CXXFLAGS) $(LDFLAGS) -o $@
+CXXLINK = $(CXX) $(CXXFLAGS) $(LDFLAGS) -o $@
 INCLUDES = -I../lib
 
 vmailmgrd_SOURCES = daemon.h \
--- ./authenticate/Makefile.am.orig	2004-05-07 10:30:39.000000000 -0400
+++ ./authenticate/Makefile.am	2004-05-07 10:30:54.000000000 -0400
@@ -20,7 +20,7 @@
 %.1: %.pod ../configure.in; $(POD2MAN) --section=1 $< >$@
 %.html: %.pod; $(POD2HTML) $< >$@
 
-CXXLINK = $(CC) $(CXXFLAGS) $(LDFLAGS) -o $@
+CXXLINK = $(CXX) $(CXXFLAGS) $(LDFLAGS) -o $@
 INCLUDES = $(CIINCLUDES) -I../lib
 
 checkvpw_SOURCES = checkvpw.cc authvlib.cc authvlib.h
--- ./authenticate/checkvpw.cc.orig	2004-05-07 10:56:06.000000000 -0400
+++ ./authenticate/checkvpw.cc	2004-05-07 10:57:26.000000000 -0400
@@ -18,6 +18,7 @@
 #include <errno.h>
 #include <string.h>
 #include <sys/types.h>
+#include <stdlib.h>
 #include "config/configrc.h"
 #include "misc/exec.h"
 #include "misc/stat_fns.h"
--- ./authenticate/vauthenticate.cc.orig	2004-05-07 11:03:27.000000000 -0400
+++ ./authenticate/vauthenticate.cc	2004-05-07 11:03:42.000000000 -0400
@@ -17,6 +17,7 @@
 #include <config.h>
 #include <errno.h>
 #include <unistd.h>
+#include <stdlib.h>
 #include "authvlib.h"
 #include "fdbuf/fdbuf.h"
 
--- ./commands/vpasswd2cdb.cc.orig	2004-05-07 11:06:27.000000000 -0400
+++ ./commands/vpasswd2cdb.cc	2004-05-07 11:06:52.000000000 -0400
@@ -27,9 +27,9 @@
 const char* cli_program = "vpasswd2cdb";
 const char* cli_help_prefix = "Converts text password tables to CDB format\n";
 const char* cli_help_suffix =
-"Reads in a standard virtual password table in the current directory,
-and writes it out to a CDB table.  The file names for the text and CDB
-tables are determined from the configuration file.";
+"Reads in a standard virtual password table in the current directory,\n"
+"and writes it out to a CDB table.  The file names for the text and CDB\n"
+"tables are determined from the configuration file.";
 const char* cli_args_usage = "";
 const int cli_args_min = 0;
 const int cli_args_max = 0;
--- ./commands/vchforwards.cc.orig	2004-05-07 11:13:22.000000000 -0400
+++ ./commands/vchforwards.cc	2004-05-07 11:14:04.000000000 -0400
@@ -25,8 +25,7 @@
 const char* cli_help_prefix =
 "Change virtual user forwarding addresses.\n";
 const char* cli_help_suffix =
-"If no forwarding addresses are given, forwarding is disabled.
-";
+"If no forwarding addresses are given, forwarding is disabled.\n";
 const char* cli_args_usage = "USERNAME [DESTINATION1 ...]";
 const int cli_args_min = 1;
 const int cli_args_max = -1;
--- ./commands/vcheckquota.cc.orig	2004-05-07 11:17:37.000000000 -0400
+++ ./commands/vcheckquota.cc	2004-05-07 11:18:11.000000000 -0400
@@ -31,10 +31,10 @@
 
 const char* cli_program = "vcheckquota";
 const char* cli_help_prefix = "vmailmgr quota enforcement program\n";
-const char* cli_help_suffix = "
-Warning: the soft-message is linked into the users maildir once for each
-message that is received while the account is over its soft quota.  This may
-result in multiple warning messages.\n";
+const char* cli_help_suffix = 
+"\nWarning: the soft-message is linked into the users maildir once for each\n"
+"message that is received while the account is over its soft quota.  This may\n"
+"result in multiple warning messages.\n";
 const char* cli_args_usage = "";
 const int cli_args_min = 0;
 const int cli_args_max = 0;
--- ./commands/vpasswd2db.cc.orig	2004-05-07 11:22:08.000000000 -0400
+++ ./commands/vpasswd2db.cc	2004-05-07 11:22:35.000000000 -0400
@@ -26,9 +26,9 @@
 const char* cli_program = "vpasswd2db";
 const char* cli_help_prefix = "Converts text password tables to current vpwtable DB format\n";
 const char* cli_help_suffix =
-"Reads in a standard virtual password table in the current directory,
-and writes it out to a table.  The file names for the input and output
-tables are determined from the configuration file.";
+"Reads in a standard virtual password table in the current directory,\n"
+"and writes it out to a table.  The file names for the input and output\n"
+"tables are determined from the configuration file.";
 const char* cli_args_usage = "";
 const int cli_args_min = 0;
 const int cli_args_max = 0;
